request/src/main/java/com/vimeo/networking2/VimeoApiClient.kt

```diff
@@ -422,6 +410,147 @@ interface VimeoApiClient {
      */
     fun deleteConnectedApp(type: ConnectedAppType, callback: VimeoCallback<Unit>): VimeoRequest
 
+    /**
+     * Create a folder that will be used to organize videos.
+     *
+     * @param uri The URI of the user's projects connection.
+     * @param name The name of the folder.
+     * @param privacy The privacy of the folder.
+     * @param callback The callback which will be notified of the request completion.
+     *
+     * @return A [VimeoRequest] object to cancel API requests.
+     */
+    fun createFolder(
+        uri: String,
+        name: String,
+        privacy: FolderViewPrivacyType,
+        callback: VimeoCallback<Folder>
+    ): VimeoRequest
+
+    /**
+     * Create a folder that will be used to organize videos.
+     *
+     * @param user The user whose projects connection will be used for the request.
+     * @param name The name of the folder.
+     * @param privacy The privacy of the folder.
+     * @param callback The callback which will be notified of the request completion.
+     *
+     * @return A [VimeoRequest] object to cancel API requests.
+     */
+    fun createFolder(
```

Comment:
The api also supports slack related parameters and a parent web folder id here. Maybe we should add them in now so we are ready to support this later? 

https://github.vimeows.com/Vimeo-API/design/blob/master/docs/3.4.2/apiblueprint/resources/Folders/Essentials.apib

Reply:
I added the slack parameters, but decided against adding the parent folder ID since that's not an available feature yet. We can add it later when nested folders is functional and the API is fully fleshed out for it.